### PR TITLE
Use descriptive variable names.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR updates much of the codebase to use longer/clearer variable names, such as `operation` instead of `op`. I aimed for ensuring consistency between different sections of the codebase, so that similarly-named variables tend to be of the same type.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Review by cbkerr — APPROVED** [review]
The changes look fine. Wondering if the following variables in directives.py should be expanded?
- line 256: `np`
- line 276 `v` and `o`
- also `_NP` and `_NP_DEFAULT`

**Comment by bdice** [comment]
@cbkerr Thanks for the review!
1. I left `np` as-is because that's a specific argument name that we pass to schedulers/mpirun during submission. It's part of our API for directives and thus changing it would be difficult.
2. Oops. I thought I had changed that one, but I was conflicted about exactly what to call it. I made it `_no_aggregation(value, other)`, which is what I'd guess the original intention was for `v, o`.
3. Leaving `_NP` and `_NP_DEFAULT` for same reasons as (1).

File: flow/project.py [hunk 1]
```diff
@@ -2835,17 +2861,17 @@ class _PickleError(Exception):
         pass
 
     @staticmethod
-    def _dumps_op(op):
+    def _operation_to_tuple(operation):
```
Comment (b-butler) [inline flow/project.py:2864]:
```suggestion
    def _job_operation_to_tuple(operation):
```

File: flow/project.py [hunk 2]
```diff
@@ -2835,17 +2861,17 @@ class _PickleError(Exception):
...
+            operation.id,
+            operation.name,
+            [job.get_id() for job in operation._jobs],
+            operation.cmd,
+            operation.directives,
         )
 
-    def _loads_op(self, blob):
-        id, name, job_ids, cmd, directives = blob
+    def _operation_from_tuple(self, data):
```
Comment (b-butler) [inline flow/project.py:2873]:
```suggestion
    def _job_operation_from_tuple(self, data):
```

File: flow/project.py [hunk 3]
```diff
@@ -2862,16 +2888,24 @@ def _run_operations_in_parallel(self, pool, pickle, operations, progress, timeou
...
-            s_tasks = [
-                (pickle.loads, s_root, self._dumps_op(op), s_ops, "run_operations")
-                for op in tqdm(operations, desc="Serialize tasks", file=sys.stderr)
+            serialized_root = pickle.dumps(self.root_directory())
+            serialized_operations = pickle.dumps(self._operations)
+            serialized_tasks = [
+                (
+                    pickle.loads,
+                    serialized_root,
+                    self._operation_to_tuple(operation),
```
Comment (b-butler) [inline flow/project.py:2897]:
```suggestion
                    self._job_operation_to_tuple(operation),
```

File: flow/project.py [hunk 4]
```diff
@@ -4888,9 +4931,9 @@ def _serializer(loads, root, *args):
     root = loads(root)
     project = FlowProject.get_project(root)
     if args[-1] == "run_operations":
-        operation = args[0]
+        operation_data = args[0]
         project._operations = loads(args[1])
-        project._execute_operation(project._loads_op(operation))
+        project._execute_operation(project._operation_from_tuple(operation_data))
```
Comment (b-butler) [inline flow/project.py:4936]:
```suggestion
        project._execute_operation(project._job_operation_from_tuple(operation_data))
```

**Review by b-butler — APPROVED** [review]
Thanks for the PR! I think we could use job_operation more frequently given the number of operation classes. However, this can be left until later.

**Comment by bdice** [comment]
@b-butler Good idea about “job operation.” I think future refactorings can improve that further.

**Pull request merged**